viewing_party/party.py

```diff
@@ -1,23 +1,138 @@
+from re import T
+
 # ------------- WAVE 1 --------------------
 
+# This function adds 3 key-value pairs to a dictionary called movie_dict
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if title and genre and rating:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict
+    else:
+        return None
+
+# This function adds a key-value pair called "watched" to a dictionary called user_data; It includes all movies that the user has watched 
+def add_to_watched(user_data, movie):
+    user_data["watched"] = [movie]
+    return user_data
+
+# This function adds a key-value pair called "watchlist" to a dictionary called user_data. It includes all movies that the user wants to watch 
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"] = [movie]
+    return user_data
+
+# This function decides whether to put a movie into a watchlist or a list of watchd movie
+def watch_movie(user_data, title):
+    for index in range(len(user_data["watchlist"])):
+        if title == user_data["watchlist"][index]["title"]:
+            watched_movie = user_data["watchlist"][index]
+            user_data["watchlist"].remove(watched_movie)
+            user_data["watched"].append(watched_movie)
+    return user_data

```

Comment:
This can be made clearer by removing the use of `range`:

```suggestion
    for watched_movie in user_data["watchlist"]:
        if title == watched_movie["title"]:
            user_data["watchlist"].remove(watched_movie)
            user_data["watched"].append(watched_movie)
    return user_data
```